sdk/pinocchio/src/program_error.rs

```diff
@@ -236,3 +232,54 @@ impl From<ProgramError> for u64 {
         }
     }
 }
+
+/// A trait for converting a program error to a `&str`.
+pub trait ToStr {
+    fn to_str<E>(&self) -> &'static str
+    where
+        E: 'static + ToStr + TryFrom<u32>;
+}
```

Comment:
Does this need the `'static` lifetime? It might be more generally useful if it has a more limited lifetime

Reply:
The issue is that it is returning a reference to a value created on the function `to_str`, so any other lifetime won't be accepted.

Reply:
Ah right, makes sense -- oh well!